ichnaea/api/views.py

```diff
@@ -140,21 +155,36 @@ def __call__(self):
                 # if we cannot connect to backend DB, skip api key check
                 skip_check = True
                 self.raven_client.captureException()
-
-        if api_key is not None and api_key.allowed(self.view_type):
+                bind_threadlocal(
+                    api_key=api_key_text,
+                    api_path=self.metric_path,
+                    api_type=self.view_type,
+                    api_key_db_fail=True,
+                )
```

Comment:
This except clause covers any kind of exception--not just a db fail. Is is prudent to restrict this further?

Reply:
I hate this style of exception handlers, but it is in the original code. I'll try to determine why this except clause was added and see if there is a more focused exception we can catch.

Reply:
There are some cases (*cough*boto*cough*) where it's really hard. I don't know what the case might be here.

Reply:
It looks like it was changed to ``Exception`` in 2014 in commit https://github.com/mozilla/ichnaea/commit/25a93524c199cfef379dc063d62de61d8fd5e305.

Looking at https://github.com/PyMySQL/PyMySQL/blob/master/pymysql/err.py and Sentry, I think PyMySQL's  ``DatabaseError`` should catch connection errors and errors due to DB reboots.

Reply:
and SQLAlchemy's OperationalError to keep the tests working

Reply:
:+1: